### PR TITLE
Fix nested mentions

### PR DESCRIPTION
**Pull request by alejandronanez**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- hyperion (frontend)

----

With this input `Inline mentions: @ale @ana @felix // @ale @ana @felix // @felix @ana @ale` users got:

## Previous output
before

After this PR users will get:

## Fixed output 🙂
good

## TODO
- [x] Unit tests
- [x] Probably some cleanup.

Closes https://github.com/withspectrum/spectrum/issues/2963

### CODE REVIEW

File: shared/clients/draft-js/mentions-decorator/core.js [hunk 1]
```diff
@@ -9,6 +9,28 @@ export type MentionComponentPropsType = {
...
+//  "Hello @ale @ange @ale @ange" should return => [[6, 9], [11, 15], [17, 20], [22, 26]]
+// 2. With that array we can call the callback function.
+export const getMentionsPositionsFromMessage = (
+  message: string
+): Array<{ startPos: number, endPos: number }> => {
+  const mentionCoordinates = [];
+  const newMessage = message;
+
+  // $FlowFix
+  newMessage.replace(MENTIONS, (match, position) => {
```
Comment (alejandronanez) [inline shared/clients/draft-js/mentions-decorator/core.js]: This is the better way I found to get the start and end position of the mentions

**Comment by alejandronanez** [comment]
Oh and BTW this is no longer happening in code blocks after https://github.com/withspectrum/spectrum/pull/2961 hit `alpha`  😄

**Review by mxstbr — COMMENTED** [review]
Looks super great! It'd be awesome to add unit tests to this whole block of behavior, although I don't know how to tackle that.

**Comment by alejandronanez** [comment]
Yup I agree about covering the whole block but I don’t know how to test it as a whole either.
I can create an e2e test for this though if you think is worth it though.

**Review by mxstbr — COMMENTED** [review]
Probably not worth it—I'd say we ship this then and figure out how to test this later?

**Comment by alejandronanez** [comment]
Yeah sounds good.
I’ll add tests to the new function I created and this will be ready to go.

**Comment by brianlovin** [comment]
Afaik that e2e failure is a random one. @mxstbr I think this is good to go for your review :)

**Review by mxstbr — APPROVED** [review]
Looks great to me! 😍Thanks so much!

**Pull request merged**